notification/notification.js

```diff
@@ -1,40 +1,189 @@
-// on swapping tab update URL
-function updateCurrentUrl() {
-	function logTabs(tabs) {
-		const tab = tabs[0]; // Safe to assume there will only be one result
-		showNotification(tab.url);
-	}
-	function onError(err){
-		showNotification(null);
-	}
-	browser.tabs.query({currentWindow: true, active: true}).then(logTabs, onError);
-}
+﻿﻿function updateCurrentUrl() {
+    function logTabs(tabs) {
+        if(tabs[0].url) {
+            handleMessage(tab[0].url);
+        }
+     }
+    function onError(err){
+        showNotification(null);
+    }
+    browser.tabs.query({currentWindow: true, active: true}).then(logTabs, onError);
+} 
 browser.tabs.onActivated.addListener(updateCurrentUrl);
-browser.tabs.onUpdated.addListener(updateCurrentUrl);
 
 //service is loaded into the page (getCurrentURL.js)
 const service = browser.runtime.connect({name:"updateWindowURL"});
+
+function findBetterAlternative(currentUrl) {
+    console.log('AlternativeJS: ', alternativeApps)
+    let betterAlternative
+    alternativeApps.softwaresList.forEach((next, key) => {
+        if(next.url[0] == currentUrl) {
+            betterAlternative = next.alternatives[0].url
+        }
+    })
+    return betterAlternative
+}
+
 function handleMessage(request) {
     currentUrl = request.currentWindowURL;
+    let betterAlternative = findBetterAlternative(currentUrl);
```

Comment:
Try and use const (instead of let) whenever possible

Reply:
I wanted to use let because it is a local way to declare a variable and it does not conflict outside. But OK.

Reply:
I was rushing and should have explained my comment better. Using let is valid here, I just suggested it since you don't change it's value anywhere in the code. 

This [answer from stackoverflow](https://stackoverflow.com/questions/41086633/why-most-of-the-time-should-i-use-const-instead-of-let-in-javascript) highlights why I consider it's best practice to use const whenever possible: 

```
You should use let or const as appropriate in your code:

* let if the variable's value will change during the code
* const if it won't (but this is a matter of style subject to your project's style guide)

In practice, if you're following the usual rules of keeping your functions small and such,
it's surprising how often it turns out that that means you use const (if you / your team want to) 
because you end up not needing to change a variable's value. (Well, it surprised me, anyway...)

Using const when the variable's¹ value is not meant to change accomplishes a few things:

* It tells others reading your code that you don't intend the value to change.
* It gives you a nice proactive error if you do something that would change the variable's value 
without having meant to do that. (A smart IDE can flag this up proactively.)
* It gives a hint to the JavaScript engine's optimizer that you won't be changing that variable's 
value. While the engine can frequently work that out through static code analysis, using const 
saves it the trouble, and lets it do that when static code analysis may not reveal that the value 
is unchanging.
* When you get the error in #2 above, you can then make an intelligent decision: 
Should you change it to let, or did you not mean to change that variable's value in the first place?```